transformer_engine/pytorch/attention.py

```diff
@@ -756,10 +771,14 @@ def forward(
         """MultiHeadAttention FWD"""
         # hidden_states: [sq, b, h]
 
-        if self.attn_mask_type != "causal" and attention_mask is not None:
+        if self.attn_mask_type == "padding":
+            assert (
+                attention_mask is not None
+            ), "Boolean attention mask must be provided for padding."
```

Comment:
The m512 version of the fused attention supports padding as well, but it doesn't require a mask tensor. Unfortunately which backend to use for fused attention is only determined later in DPA forward. So maybe we can move this check to the FlashAttention module instead.

Reply:
How does padding work without a mask tensor provided?